### PR TITLE
Add HiveMetadataUpdater to handle metadata updates

### PR DESCRIPTION
**Pull request by NikhilCollooru**
```
== NO RELEASE NOTE ==

```

depended by https://github.com/facebookexternal/presto-facebook/pull/1156

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadataUpdateHandle.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import java.util.UUID;
+
+import static java.util.Objects.requireNonNull;
+
+public class HiveMetadataUpdateHandle
+        implements ConnectorMetadataUpdateHandle
+{
+    private final UUID requestId;
+    private final String schemaName;
+    private final String schemaTableName;
```
Comment (highker) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadataUpdateHandle.java:31]: What I meant was to use `SchemaTableName` type from SPI

Comment (NikhilCollooru) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadataUpdateHandle.java]: oops...didnt know it existed. Made the suggested change.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadataUpdater.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import java.util.Queue;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.Executor;
+
+import static java.util.Objects.requireNonNull;
+
+public class HiveMetadataUpdater
```
Comment (highker) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadataUpdater.java:35]: Add to TODO: this class requires visit to make it more robust

File: presto-main/src/main/java/com/facebook/presto/operator/TableWriterOperator.java [hunk 3]
```diff
@@ -136,6 +162,19 @@ private ConnectorPageSink createPageSink()
             throw new UnsupportedOperationException("Unhandled target type: " + target.getClass().getName());
         }
 
+        private ConnectorId getConnectorId(ExecutionWriterTarget handle)
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/operator/TableWriterOperator.java:165]: static

File: presto-main/src/main/java/com/facebook/presto/operator/TableWriterOperator.java [hunk 4]
```diff
@@ -124,9 +135,24 @@ public Operator createOperator(DriverContext driverContext)
...
+            TaskMetadataContext metadataContext;
+            Optional<ConnectorMetadataUpdater> metadataUpdater = metadataUpdaterManager.getMetadataUpdater(connectorId);
+            if (metadataUpdater.isPresent()) {
+                if (!taskContext.getTaskMetadataContext().isPresent()) {
+                    metadataContext = new TaskMetadataContext(connectorId);
+                    taskContext.setTaskMetadataContext(metadataContext);
+                }
+
+                metadataContext = taskContext.getTaskMetadataContext().get();
+                metadataContext.addMetadataUpdater(metadataUpdater.get());
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/operator/TableWriterOperator.java:148]: Not a big fan of having setter and getter here. If we have to have something mutable, I would say make metadataContext in taskContext immutable and connectorId in metadataContext mutable. That will make this cleaner.

**Review by highker — APPROVED** [review]
Already reviewed in another PR; minor comments only.

File: presto-main/src/main/java/com/facebook/presto/operator/TableWriterOperator.java (hunk 4 above)
Comment (NikhilCollooru) [reply presto-main/src/main/java/com/facebook/presto/operator/TableWriterOperator.java]: Done. Made the `ConnectorId` in `TaskMetadataContext` mutable

**Pull request merged**